src/components/shared/modal/modal.tsx

```diff
@@ -0,0 +1,26 @@
+import type { ComponentChildren } from "preact";
+import "./modal.css";
+
+type Props = {
+  title: string;
+  className?: string;
+  children: ComponentChildren;
+  onTitleClick?: () => void;
+};
+
+export function ModalComponent({
+  title,
+  children,
+  className,
+  onTitleClick,
+}: Props) {
+  const classString = className ? `modal ${className}` : "modal";
```

Comment:
Maybe you didn't know) There is classname npm package specifically for this cases. You can use it. But maybe you just didn't because it's only few places where it's needed so it's unnecessary to download whole package for that.

Reply:
Yes, I wanted to use it, but it's larger than Preact + signals. So, I decided not to use it. :)